src/Orleans/Serialization/IlDelegateBuilder.cs

```diff
@@ -81,6 +77,62 @@ public void LoadArgument(ushort index)
         }
 
         /// <summary>
+        /// Loads the element from the array on the stack at the given index onto the stack.
+        /// </summary>
+        public void LoadReferenceElement()
+        {
+            this.il.Emit(OpCodes.Ldelem_Ref);
+        }
+
+        /// <summary>
+        /// Loads the provided constant integer value onto the stack.
+        /// </summary>
+        public void LoadConstant(int value)
+        {
+            switch (value)
+            {
+                case 0:
+                    this.il.Emit(OpCodes.Ldc_I4_0);
+                    break;
+                case 1:
+                    this.il.Emit(OpCodes.Ldc_I4_1);
+                    break;
+                case 2:
+                    this.il.Emit(OpCodes.Ldc_I4_2);
+                    break;
+                case 3:
+                    this.il.Emit(OpCodes.Ldc_I4_3);
+                    break;
+                case 4:
+                    this.il.Emit(OpCodes.Ldc_I4_4);
+                    break;
+                case 5:
+                    this.il.Emit(OpCodes.Ldc_I4_5);
+                    break;
+                case 6:
+                    this.il.Emit(OpCodes.Ldc_I4_6);
+                    break;
+                case 7:
+                    this.il.Emit(OpCodes.Ldc_I4_7);
+                    break;
+                case 8:
+                    this.il.Emit(OpCodes.Ldc_I4_8);
```

Comment:
Yep, and they're smaller. It's just an optimization, so I can remove it if preferred.